### DISCUSSION TITLE
Support multiple scenarios for a given topology file

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#66] [Ideas]
While working on [BGP Add path using SROS](https://github.com/jbemmel/netsim-examples/tree/jvb_add_sros_srlinux_bgp_addpath_example/BGP/Multipath_sros) it occurred to me that it would be nice if one could create a single topology file to evaluate multiple different solutions/options

In my case, I would have 2 scenarios:
1. BGP Add path, using SROS for "RR" and "M" nodes (because those support/require that capability)
2. BGP Anycast, using SROS for "C" and "D" nodes (because of the need to set next-hop to an arbitrary IP, not supported by all platforms)

The device topology and links would be the same, just a different combination of vendors/OS choices for different nodes.

Of course one can always create 2 topology files, but that disregards a main intent (evaluating several alternatives against the same setup)

To implement this, it would be nice if 'netlab create' would put its generated files in a sub-directory (per scenario), instead of "polluting" the main directory

**Comment by ipspace** [comment]
I totally understand the use case, but I hate throwing too many nerd knobs at netlab CLI. What you're looking for is really easy to do:

```
rm -fr scenario
mkdir scenario
chdir scenario
netlab create ../topology.yml
```

Making a bash script that would do the same (plus set some variables) is also trivial. Why should a tool reinvent an easy-to-implement wheel?

It might make sense though to accept output directory in an environment variable (and **chdir** to it before calling output modules).

Thoughts?

**Comment by ipspace** [comment]
Solved with **netlab up** `-s` parameter
